conn.go

```diff
@@ -584,6 +617,60 @@ type callReq struct {
 	timer *time.Timer
 }
 
+type writeCoalescer struct {
+	w       io.Writer
+	timeout time.Duration
+
+	cond    *sync.Cond
+	mu      sync.Mutex
+	buffers net.Buffers
+
+	// result of the write
+	err error
+}
+
+func (w *writeCoalescer) flush() {
+	if w.timeout > 0 {
+		type deadliner interface {
+			SetWriteDeadline(time.Time) error
+		}
+		w.w.(deadliner).SetWriteDeadline(time.Now().Add(w.timeout))
```

Comment:
Do we have to create a custom interface here? Can we just do away with `w.w.(net.Conn)...` ? We don't really use it anywhere else, and this is all internal anyway.